habitat/tasks/rearrange/rearrange_sim.py

```diff
@@ -261,39 +276,69 @@ def reconfigure(self, config: Config):
                 ao: ao.joint_positions for ao in self.art_objs
             }
 
+    def set_robot_base_to_random_point(self):
+        for _ in range(50):
+            start_pos = self.pathfinder.get_random_navigable_point()
+
+            start_pos = self.safe_snap_point(start_pos)
+            start_rot = np.random.uniform(0, 2 * np.pi)
+
+            self.robot.base_pos = start_pos
+            self.robot.base_rot = start_rot
+
+            self.internal_step(-1)
+            did_collide, details = rearrange_collision(
+                self,
+                True,
+                ignore_base=False,
+            )
+            if not did_collide:
+                break
+
+    def _setup_targets(self):
+        self._targets = {}
+        for target_handle, transform in self.ep_info["targets"].items():
+            self._targets[target_handle] = mn.Matrix4(
+                [[transform[j][i] for j in range(4)] for i in range(4)]
+            )
+
     def _recompute_navmesh(self):
-        """Generates the navmesh on the fly. This must be called
-        AFTER adding articulated objects to the scene.
-        """
+        scene_name = self.ep_info["scene_id"].split("/")[-1].split(".")[0]
+        base_dir = osp.join(*self.ep_info["scene_id"].split("/")[:2])
+
+        navmesh_path = osp.join(base_dir, "navmeshes", scene_name + ".navmesh")
+        self.pathfinder.load_nav_mesh(navmesh_path)
 
-        # cache current motiontype and set to STATIC for inclusion in the NavMesh computation
-        motion_types = []
-        for art_obj in self.art_objs:
-            motion_types.append(art_obj.motion_type)
-            art_obj.motion_type = MotionType.STATIC
-        # compute new NavMesh
-        self.recompute_navmesh(
-            self.pathfinder,
-            self.navmesh_settings,
-            include_static_objects=True,
+        self._navmesh_vertices = np.stack(
+            self.pathfinder.build_navmesh_vertices(), axis=0
```

Comment:
This is now just loading the navmesh, not recomputing. This isn't necessary if you loaded the scene instance with a linked NavMesh unless you've manually changed the NavMesh already elsewhere. 

What if the NavMesh was not provided, should it be recomputed?

Reply:
I will change the method name. What do you mean by linked navmesh? Also, I am not sure how the user could "provide or not provide" a navmesh? Shouldn't the navmesh always be automatically looked up?

Reply:
By linked NavMesh, I mean a `.navmesh` file was provided in the SceneDataset and referenced in the `.scene_instance_config.json`. This case is automatically handled by the C++ sim code and setup for ReplicaCADv1.4, so it makes the logic here redundant.

However, in general, there may be cases where rearrange will be run on other datasets which may not have NavMesh files (or you may not know where to find them), so recomputing the NavMesh manually should still be a backup. I guess this may also be done automatically in the Simulator init, but any specific settings (e.g. for Fetch base radius) should maybe be included here.

Reply:
But I am not building the navmesh, I am loading the specified name. What changes would you suggest to this method?